pkg/controller/argocd/secret.go

```diff
@@ -379,6 +385,38 @@ func (r *ReconcileArgoCD) reconcileGrafanaSecret(cr *argoprojv1a1.ArgoCD) error
 	return r.client.Create(context.TODO(), secret)
 }
 
+// reconcileClusterPermissionsSecret ensures ArgoCD instance is namespace-scoped
+func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.ArgoCD) error {
```

Comment:
I'm thinking this method should handle a scenario where user already has created a cluster secret for the `in-cluster` cluster, but with another name. Should we look for secrets in the current namespace with a label `argocd.argoproj.io/secret-type` and `.data.name` value of `in-cluster`?

I'm not quite sure how Argo CD would behave when there are two cluster with the same value of `.data.name`. Probably, the behaviour would be undefined and it would pick randomly the first one returned by the K8s API.

Reply:
makes sense, i'll try this scenario out. Thanks !!

Reply:
done!